packages/react/README.md

```diff
@@ -101,3 +119,593 @@ function WidgetDeleter() {
+
+`useFindFirst` returns two values: a result object with the `data`, `fetching`, and `error` keys for inspecting in your React component's output, and a `refetch` function to trigger a refresh of the hook's data.
+
+- `data: GadgetRecord | null`: The record fetched from the backend. Is `null` while the data is being loaded, or if a matching record wasn't found.
+- `fetching: boolean`: A boolean describing if the hook is currently making a request to the backend.
+- `error: Error | null`: An error from the client or server side, if encountered during the request. Will contain an error if the first record isn't found. See the [Errors](#errors-from-the-returned-error-object) section.
+
+Without any options, `useFindFirst` will fetch the first matching record and cause your component to rerender as the fetch happens and when the data or error arrives:
+
+```javascript
+import React from "react";
+import { api } from "../api"; // some file that instantiates @gadget-client/your-app-slug
+
+export const MostRecentPublishedPost = (props: { id: string }) => {
+  // request the first record from the backend where publishedAt is set and with the most recent publishedAt value
+  const [{ data, fetching, error }, _refetch] = useFindFirst(api.blogPost, {
+    filter: { publishedAt: { isSet: true } },
+    sort: { publishedAt: "Descending" },
+  });
+
+  if (fetching) return "Loading...";
+  if (error) return `Error loading data: ${error}`;
+  return <span className="banner">Check out our most recent blog post titled {data.title}</span>;
+};
+```
+
+`useFindFirst` can take options which allow the customization of which fields are returned:
+
+```javascript
+// fetch only the widget id and name fields
+const [{ data, fetching, error }, _refetch] = useFindOne(api.widget, props.id, { select: { id: true, name: true } });
+```
+
+The `refetch` function returned as the second element can be executed in order to trigger a refetch of the most up to date data from the backend. See [`urql`'s docs](https://formidable.com/open-source/urql/docs/basics/react-preact/#reexecuting-queries) on re-executing queries for more information.
+
+### `useFindBy(findFunction: ModelFindFunction, fieldValue: any, options: FindByOptions = {}): [{data, fetching, error}, refetch]`
+
+`useFindBy` fetches one record from your backend Gadget, looked up by a specific field and value. `useFindBy` uses the existing specific-field finder functions generated within your application's API client, like `api.user.findByEmail` or similar. `useFindBy` expects to be passed a field-specific finder function as the first argument, and then the value of the field to find by as the second argument, and then optionally accepts a third options object. If no record is found matching the conditions, then the returned object will have `null` for the `data`. `useFindBy(api.widget.findByEmail, "hi@gadget.dev")` is the React equivalent of `api.widget.findByEmail("hi@gadget.dev")`. Options:
```

Comment:
Do we want to call out here that these are only generated in your client for fields that have a uniqueness validation?